From: Michael Tautschnig <mt@debian.org>
Date: Sat, 28 Jun 2014 20:48:21 +0100
Subject: Abort on invalid header

Upstream has made a different change to validate input.

Bug-Debian: https://bugs.debian.org/729274
Forwarded: not-needed
Applied-Upstream: commit:75b8939e07dc255a6b075be0fd71b0d03bec1efe
---
 bfd/tekhex.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bfd/tekhex.c b/bfd/tekhex.c
index df1dab5..ecb27b4 100644
--- a/bfd/tekhex.c
+++ b/bfd/tekhex.c
@@ -330,6 +330,7 @@ getsym (dstp, srcp)
   char *src = *srcp;
   unsigned int i;
   unsigned int len = hex_value(*src++);
+  if(len>=16) abort();
 
   if (len == 0)
     len = 16;
